pkg/trace/transform/transform.go

```diff
@@ -210,14 +218,22 @@ func MarshalEvents(events ptrace.SpanEventSlice) string {
 			str.WriteString(`"attributes":{`)
 			j := 0
 			e.Attributes().Range(func(k string, v pcommon.Value) bool {
-				if j > 0 {
-					str.WriteString(",")
+				// collect the attribute only if the key is json-parseable, else drop the attribute
+				if key, err := json.Marshal(k); err == nil {
+					if j > 0 {
+						str.WriteString(",")
+					}
+					str.WriteString(string(key))
+					str.WriteString(":")
+					if val, err := json.Marshal(v.AsString()); err == nil {
+						str.WriteString(string(val))
+					} else {
+						log.Warnf("Trouble parsing the following attribute value, dropping: %v", v.AsString())
+						str.WriteString(`"redacted"`)
+					}
```

Comment:
nit: I think the j++ needs to be moved up here to indicate that a value was written

```suggestion
					}
					j++
```